### PR TITLE
Simplify cm::vhost cm::reverse_proxy UX

### PR DESCRIPTION
**Pull request by ppp0**
via #1628

### CODE REVIEW

**Comment by ppp0** [comment]
@kris-lab please review

File: modules/cm/manifests/reverse_proxy.pp [hunk 1]
```diff
@@ -65,9 +52,9 @@
       'real_ip_recursive on;',
       'real_ip_header X-Real-IP;',
       'set_real_ip_from 0.0.0.0/0;',
-      "proxy_set_header Host '${upstream_opts[header_host]}';",
+      'proxy_set_header Host $host;',
       'proxy_set_header X-Real-IP $remote_addr;',
-      "proxy_pass ${proto}://${upstream_name_real};",
+      "proxy_pass ${upstream_protocol}://${backend_name};",
```
Comment (kris-lab) [inline modules/cm/manifests/reverse_proxy.pp]: I am wondering how it works if you specify proxy protocol as `https`, but upstream members are mix of `80/http` and `443/https`, do you know?

File: modules/cm/manifests/vhost.pp [hunk 2]
```diff
@@ -1,38 +1,30 @@
...
-  $aliases = [],
-  $redirects = undef,
-  $cdn_origin = undef,
-  $debug = false,
-  $upstream_options = { },
+  $ssl_port          = 443,
+  $aliases           = [],
+  $redirects         = undef,
+  $cdn_origin        = undef,
+  $path,
```
Comment (kris-lab) [inline modules/cm/manifests/vhost.pp]: Shouldn't we move all required params on the top of list?

**Comment by kris-lab** [comment]
I like it and just wondering if we need to specify `upstream_protocol` and how it works with mixed protocols?

File: modules/cm/manifests/reverse_proxy.pp (hunk 1 above)
Comment (ppp0) [inline modules/cm/manifests/reverse_proxy.pp]: I don't think we have a use case in `cm::` for this at the moment. Is it possible at all? That would either require several `proxy_pass` in the same `location` context. Even if this is possible (may be by defining a mix of http and https backends in `upstream` it still looks to me like a hypothetical use case. No?

File: modules/cm/manifests/vhost.pp (hunk 2 above)
Comment (ppp0) [inline modules/cm/manifests/vhost.pp]: makes sense!

**Comment by ppp0** [comment]
@kris-lab  please have a look again

File: modules/cm/manifests/reverse_proxy.pp (hunk 1 above)
Comment (kris-lab) [inline modules/cm/manifests/reverse_proxy.pp]: Agree, we do not implement such case for the moment.

**Comment by kris-lab** [comment]
I think it good. Let's wait for build and apply carefully.

**Pull request merged**